stress/common/src/main/java/alluxio/stress/client/ClientIOTaskResult.java

```diff
@@ -0,0 +1,344 @@
+      List<ClientIOTaskResult> summaries =
+          results.stream().map(x -> (ClientIOTaskResult) x).collect(Collectors.toList());
+
+      // Iterate over all operations
+      for (Operation operation : Operation.values()) {
+        List<ClientIOTaskResult> opSummaries =
+            summaries.stream().filter(x -> x.mParameters.mOperation == operation)
+                .collect(Collectors.toList());
+
+        if (!opSummaries.isEmpty()) {
+          // first() is the list of common field names, second() is the list of unique field names
+          Pair<List<String>, List<String>> fieldNames = Parameters.partitionFieldNames(
+              opSummaries.stream().map(x -> x.mParameters).collect(Collectors.toList()));
+
+          // Split up common description into 100 character chunks, for the sub title
+          List<String> subTitle = new ArrayList<>(Splitter.fixedLength(100).splitToList(
+              opSummaries.get(0).mParameters.getDescription(fieldNames.getFirst())));
+
+          for (ClientIOTaskResult summary : opSummaries) {
+            String series = summary.mParameters.getDescription(fieldNames.getSecond());
+            subTitle.add(series + ": " + DateFormat.getDateTimeInstance()
+                .format(summary.computeLastEndMs()));
+          }
+
+          LineGraph responseTimeGraph =
+              new LineGraph(operation + " - Throughput", subTitle, "# Threads",
+                  "Throughput (MB/s)");
+
+          for (ClientIOTaskResult summary : opSummaries) {
+            String series = summary.mParameters.getDescription(fieldNames.getSecond());
+            responseTimeGraph.addDataSeries(series, summary.getThroughputData());
+            responseTimeGraph.setErrors(series, summary.collectErrors());
+          }
+
+          graphs.add(responseTimeGraph);
+        }
+      }
+
+      return graphs;
+    }
+  }
+
+  /**
+   * A result for a single thread count test.
+   */
+  public static final class ThreadCountResult {
+    private long mRecordStartMs;
+    private long mEndMs;
+    private long mDurationMs;
+    private long mIOBytes;
+    private float mIOMBps;
+    private List<String> mErrors;
+
+    /**
+     * Creates an instance.
+     */
+    public ThreadCountResult() {
+      // Default constructor required for json deserialization
+      mErrors = new ArrayList<>();
+    }
+
+    /**
+     * Merges (updates) a result with this result.
+     *
+     * @param result  the result to merge
+     */
+    public void merge(ClientIOTaskResult.ThreadCountResult result) {
+      mRecordStartMs = result.mRecordStartMs;
+      mEndMs = Math.max(mEndMs, result.mEndMs);
+      mIOBytes += result.mIOBytes;
+      mErrors.addAll(result.mErrors);
+      update();
+    }
+
+    /**
+     * Update various measurements given the existing measurements.
+     */
+    public void update() {
+      mDurationMs = mEndMs - mRecordStartMs;
+      mIOMBps = ((float) mIOBytes / mDurationMs) * 1000.0f / Constants.MB;
```

Comment:
I would avoid making these two variables instance variables and just have getters for them that do this. Even if this thing is being json serialized, it's not like your json deserialization is going to use these variables correctly if they aren't these values.

Reply:
Done.